sklearn/utils/validation.py

```diff
@@ -339,8 +352,14 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None,
     X_converted : object
         The converted and validated X.
     """
-    if isinstance(accept_sparse, str):
-        accept_sparse = [accept_sparse]
+    # accept_sparse 'None' deprecation check
+    if accept_sparse is None:
+        warnings.warn(
+                "Passing 'None' to parameter 'accept_sparse' is "
```

Comment:
Add the function name to the warning. The origin of the warning is sometimes not trivial to understand so you want to make it as explicit as possible.